targets/TARGET_STM/trng_api.c

```diff
@@ -42,23 +42,48 @@ void trng_init(trng_t *obj)
         error("Only 1 RNG instance supported\r\n");
     }
 
-#if !defined(TARGET_STM32WB)
-    /*  Because M0 core of WB also needs RG RNG is already clocked by default */
-#if defined(RCC_PERIPHCLK_RNG)
+#if defined(RCC_PERIPHCLK_RNG) /* STM32L4 / STM32H7 / STM32WB */
+
+#if defined(TARGET_STM32WB)
+    /*  No need to reconfigure RngClockSelection as RNG is already clocked by M0 */
+
+#elif defined(TARGET_STM32H7)
     RCC_PeriphCLKInitTypeDef PeriphClkInitStruct;
 
     /*Select PLLQ output as RNG clock source */
     PeriphClkInitStruct.PeriphClockSelection = RCC_PERIPHCLK_RNG;
-#if ((CLOCK_SOURCE) & USE_PLL_MSI)
-    PeriphClkInitStruct.RngClockSelection = RCC_RNGCLKSOURCE_MSI;
-#else
     PeriphClkInitStruct.RngClockSelection = RCC_RNGCLKSOURCE_PLL;
-#endif
     if (HAL_RCCEx_PeriphCLKConfig(&PeriphClkInitStruct) != HAL_OK) {
         error("RNG clock configuration error\n");
     }
+
+#elif defined(TARGET_STM32L4)
+    /* RNG clock configuration could be already done by USB */
```

Comment:
to be very precise it's not RNG clock configuration but RNG clock source selection 
and comment should add that RNG and USB clocks have the same source, so the common source selection could be already done by USB